islandora_solr_config/includes/rss_results.inc

```diff
@@ -191,7 +190,85 @@ class IslandoraSolrResultsRSS extends IslandoraSolrResults {
     $result['url'] = $base_url;
     $result['description'] = t('Aggregated search results of: @query', array('@query' => $query));
     $result['langcode'] = NULL;
+    $feed_url = request_uri();
+    $page_params = drupal_get_query_parameters();
+
+    if (!isset($page_params['page'])) {
+      $page_number = '0';
+    }
+    else {
+      $page_number = $page_params['page'];
+    }
+    if (isset($page_params['page'])) {
+      $first_params = $page_params;
+      unset($first_params['page']);
+      $link_first = url(
+        request_path(),
+        array(
+          'absolute' => TRUE,
+          'query' => $first_params,
+        )
+      );
+    }
+    else {
+      $link_first = $base_url . $feed_url;
```

Comment:
What is this trying to do? I understand is as: If there is no page passed to the search than the first page is this page. In that case would it make sense to instead just not include the the first page link at all?

Reply:
I think it's helpful to have the "first page" link available regardless; is there any reason to exclude it?

Also, the "previous" link uses `$link_first` if you're on the second page (page 1), to generate a link without a "page" parameter.

Reply:
So the problem I have with this line at the moment is that I can't use the URL function without it encoding the request parameters. Any idea what's happening there?